aspnetcore/blazor/components/index.md

```diff
@@ -1509,6 +1509,72 @@ const rootComponent = await window.Blazor.rootComponents.add(...);
 rootComponent.dispose();
 ```
 
+<xref:Microsoft.AspNetCore.Components.Web.JSComponentConfigurationExtensions.RegisterForJavaScript%2A> includes an overload that accepts a JS function to register the custom element (`javaScriptInitializer`). The following example demonstrates the dynamic registration of a component that renders a quote.
+
+> [!IMPORTANT]
+> Don't confuse the `javaScriptInitializer` parameter of <xref:Microsoft.AspNetCore.Components.Web.JSComponentConfigurationExtensions.RegisterForJavaScript%2A> with [JavaScript initializers](xref:blazor/fundamentals/startup#javascript-initializers). The name of the parameter and the JS initializers feature is coincidental.
+
+`Shared/Quote.razor`:
+
+```razor
+<div id="quoteDiv" class="m-5 p-5">
+    <h2>Quote</h2>
+    <p>@Text</p>
+</div>
+
+@code {
+    [Parameter]
+    public string? Text { get; set; }
+}
+```
+
+In `Program.cs`, add the namespace for the location of the component. The following example assumes that the `Quote` component is in the app's `Shared` folder, and the app's namespace is `BlazorSample`:
+
+```csharp
+using BlazorSample.Shared;
+```
+
+In `Program.cs`, register the `Quote` component.
+
+Blazor Server:
+
+```csharp
+options.RootComponents.RegisterForJavaScript<Quote>(identifier: "quote", 
+    javaScriptInitializer: "initializeComponent");
+```
+
+Blazor WebAssembly:
+
+```csharp
+builder.RootComponents.RegisterForJavaScript<Quote>(identifier: "quote", 
+    javaScriptInitializer: "initializeComponent");
+```
+
+In HTML:
+
+```html
+<button onclick="showQuote()">Show Quote</button>
+
+<div id="quoteContainer"></div>
+
+<script>
+  async function showQuote() {
+    let targetElement = document.getElementById('quoteContainer');
+    await Blazor.rootComponents.add(targetElement, 'quote', 
+    {
+        text: "Crow: I have my doubts that this movie is actually 'starring' " +
+        "anybody. More like, 'camera is generally pointed at.'"
+    });
+  }
+
+  async function initializeComponent() {
```

Comment:
```suggestion
  function initializeComponent() {
```

Reply:
Might also be nice to have this function accept arguments and log their values, similar to what we do [here](https://github.com/dotnet/aspnetcore/blob/1dd6f451a0df2ee8e4575dd8dca9df77fda129b1/src/Components/test/testassets/BasicTestApp/wwwroot/js/jsRootComponentInitializers.js#L5).

Reply:
Also, does this example work? We find the `javaScriptInitializer` function by looking for it on the `window` object. Either way, could we modify this slightly, so we directly attach the function to the `window` object similar to what we do in the code I linked above?

Reply:
> does this example work?

👍 Yes, it does work.

> We find the javaScriptInitializer function by looking for it on the window object. Either way, could we modify this slightly, so we directly attach the function to the window object similar to what we do in the code I linked above?

Sure ... I'll follow your advice on that. I'll get back on this first thing Friday morning.